client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -108,6 +108,17 @@ public void readerOffline(String readerId, Position lastPosition) {
 
     @Override
     public CompletableFuture<Checkpoint> initiateCheckpoint(String checkpointName, ScheduledExecutorService backgroundExecutor) {
+
+        synchronizer.fetchUpdates();
+        int maxOutstandingCheckpointRequest = this.synchronizer.getState().getConfig().getMaxOutstandingCheckpointRequest();
+        int currentOutstandingCheckpointRequest = synchronizer.getState().getCheckpointState().getUnCheckpointedHostsSize();
```

Comment:
The most clear way to count the number of outstanding checkpoints is to count the `checkpoints` list, not the `uncheckpointedHosts` map.